lib/params.js

```diff
@@ -1,44 +1,51 @@
-var fs = require('fs')
-var path = require('path')
+const fs = require('fs')
+const path = require('path')
+const createStylelint = require('stylelint').createLinter
+const editorconfig = require('editorconfig')
+const getProperty = require('./util').getProperty
+const defaultIndentWidth = ' '.repeat(2)
+const stylelintCache = new Map()
 
-var cosmiconfig = require('cosmiconfig')
-var editorconfig = require('editorconfig')
-var objectAssign = require('object-assign')
-var resolveFrom = require('resolve-from')
-
-var isEmptyObject = require('./util').isEmptyObject
-
-var defaultIndentWidth = ' '.repeat(2)
-
-function hasScssInWorkingDir (wd) {
+function hasScssInDir (wd) {
   var dirs = fs.readdirSync(wd)
   return dirs.some(function (dir) {
     return dir.match(/.*\.scss/)
   })
 }
 
-function getIndentationFromEditorConfig (from, hasScss) {
-  var config
-  if (hasScss) {
-    config = editorconfig.parseSync(path.resolve(process.cwd(), '*.scss'))
-  } else {
-    config = editorconfig.parseSync(path.resolve(process.cwd(), '*.css'))
+function getEditorConfig (file, rules) {
+  rules = rules || {}
+  if(!file || ["indentation", "no-missing-end-of-source-newline", "no-eol-whitespace"].every(function (key) {
```

Comment:
This function is to read configrations of editorconfig, so I'd not like to do this here.